ipatests/test_integration/test_trust.py

```diff
@@ -993,3 +1036,248 @@ def test_server_option_with_unreachable_ad(self):
+        Test for non default primary group
+        """
+        nonposixuser = "nonposixuser1@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group("true")
+        tasks.clear_sssd_cache(self.master)
+        self.clients[0].run_command(['id', nonposixuser])
+        # Remove trust
+        self.remove_trust(self.ad)
+
+
+class TestPosixAutoPrivateGroup(BaseTestTrust):
+    """
+    Tests for auto-private-groups option with non posix AD trust
+    """
+    topology = 'line'
+    num_ad_domains = 1
+    num_clients = 1
+
+    def test_add_posix_trust(self):
+        self.add_trust('ipa-ad-trust-posix')
+
+    @pytest.mark.xfail(reason='bz2010512', strict=True)
+    def test_gidnumber_not_coresponding_existing_group(self):
+        """
+        Test checks that AD users which contain posix attributes
+        (uidNumber and gidNumber) but there is no group with the
+        corresponding gidNumber.
+        """
+        posixuser = "testuser2@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group("true")
+        tasks.clear_sssd_cache(self.master)
+        (uid, gid) = self.get_user_id(self.clients[0], posixuser)
+        assert uid == gid
+
+    def test_uid_number_auto_private_group_false(self):
+        """
+        Test checks that posix user with only uidNumber defined
+        and gidNumber not set, is not resolvable when auto-private-group
+        is set to False
+        """
+        posixuser = "testuser1@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group("false")
+        tasks.clear_sssd_cache(self.master)
+        result = self.clients[0].run_command(['id', posixuser],
+                                             raiseonerr=False)
+        assert 'no such user' in result.stderr_text
+
+    def test_uid_number_auto_private_group_true(self):
+        """
+        Test checks that posix user with only uidNumber defined
+        and gidNumber not set, a private group is created with
+        gidNumber = uidNumber
+        """
+        posixuser = "testuser1@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group("true")
+        tasks.clear_sssd_cache(self.master)
+        tasks.clear_sssd_cache(self.clients[0])
+        (uid, gid) = self.get_user_id(self.clients[0], posixuser)
+        assert uid == gid
+
+    def test_uid_number_auto_private_group_hybrid(self):
+        """
+        Test checks that posix user with only uidNumber defined
+        and gidNumber not set, is not resolvable when auto-private-group
+        is set to hybrid
+        """
+        posixuser = "testuser1@%s" % self.ad_domain
+        self.mod_idrange_auto_private_group("hybrid")
+        tasks.clear_sssd_cache(self.master)
+        tasks.clear_sssd_cache(self.clients[0])
+        result = self.clients[0].run_command(['id', posixuser],
+                                             raiseonerr=False)
+        assert 'no such user' in result.stderr_text
+        # Remove trust
+        self.remove_trust(self.ad)
```

Comment:
We are missing tests with a posix user with uid, gid combined with the 3 possible values of auto private groups.